src/config.ts

```diff
@@ -28,7 +28,28 @@ function validateApiVersion(apiVersion: string) {
   }
 }
 
-export const validateApiPerspective = function validateApiPerspective(perspective: string) {
+export const validateApiPerspective = function validateApiPerspective(perspective: unknown) {
+  if (Array.isArray(perspective)) {
+    for (const perspectiveValue of perspective) {
+      if (perspectiveValue === 'published') {
+        continue
+      }
+      if (perspectiveValue === 'drafts') {
+        continue
+      }
+      if (
+        typeof perspectiveValue === 'string' &&
+        perspectiveValue.startsWith('r') &&
+        perspectiveValue !== 'raw'
+      ) {
+        continue
+      }
+      throw new TypeError(
+        'Invalid API perspective value, expected `published`, `drafts` or a valid release identifier string',
+      )
+    }
+    return
+  }
   switch (perspective as ClientPerspective) {
     case 'previewDrafts':
```

Comment:
I'd suggest also allowing `drafts` as a string parameter.

Reply:
Ah, yes, good point. Added in https://github.com/sanity-io/client/commit/9f105c259f6ff2c41133fa9255f6b8cfbc77e57d